src/content/1.7/development/architecture/migration-guide/forms/CRUD-form.md

```diff
@@ -0,0 +1,245 @@
+
+### Using Form builder
+
+In most cases you only need to define form builder as a service:
+
+```yaml
+#src/PrestaShopBundle/Resources/config/services/core/form/form_builder.yml
+
+  prestashop.core.form.identifiable_object.builder.contact_form_builder:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Builder\FormBuilder'
+    factory: 'prestashop.core.form.builder.form_builder_factory:create'
+    arguments:
+      - 'PrestaShopBundle\Form\Admin\Configure\ShopParameters\Contact\ContactType'
+      - '@prestashop.core.form.identifiable_object.data_provider.contact_form_data_provider'
+```
+
+{{% notice note %}}
+ Make sure you pass your form type as a string.
+{{% /notice %}}
+
+Finally, use it in your controller:
+
+```php
+    public function editAction($contactId)
+    {
+        $contactFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.contact_form_builder');
+        $contactForm = $contactFormBuilder->getFormFor($contactId);
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/Contact/Contacts/edit.html.twig', [
+            'contactForm' => $contactForm->createView(),
+        ]);
+    }
+```
+
+and the result with some common twig structure applied for identifiable object form pages looks like this:
+
+{{< figure src="../img/contact-form-rendered.png" title="Result of form builder" >}}
+
+## Form data handler
+
+Form data handler is responsible for your form data submitting and it is used by these methods:
+
+* **create($data)** - used for creating new identifiable object.
+
+* **update($id, $data)** - used for updating identifiable object.
+
+### Creating form data handler
+
+You don’t have to create the Form data handler by yourself but instead rely on `\PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\FormDataHandlerInterface`
+
+```php
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+final class ContactFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        // handle creation logic here
+        return 'my required data to use after identifiable object is created';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($id, array $data)
+    {
+        /** @var ContactId $contactId */
+        $contactId = $this->updateContact($id, $data);
+
+        return $contactId->getValue();
+    }
+
+    public function updateContact($id, $data)
+    {
+        // Identifiable object update logic goes here.
+    }
+}
+```
+
+After initializing `create` method there you must handle identifiable objects creation logic. You can return any data you like or no data at all - it depends on the use case where the data will be used next.
+After initializing `update` method you must update your identifiable object and it must return an `int` type id.
```

Comment:
`After initializing ...` you dont really initialize it, its a simple method call. what is more, you dont even have to call it yourself since `FormHandler` does that for you. I think this explanation should be rephrased.